api/v1alpha1/protectedvolumereplicationgrouplist_types.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2022 The RamenDR authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ProtectedVolumeReplicationGroupListSpec defines the desired state of ProtectedVolumeReplicationGroupList
+type ProtectedVolumeReplicationGroupListSpec struct {
+	// ProfileName is the name of the S3 profile in the Ramen operator config map
+	// specifying the store to be queried
+	S3ProfileName string `json:"s3ProfileName"`
+}
+
+// ProtectedVolumeReplicationGroupListStatus defines the observed state of ProtectedVolumeReplicationGroupList
+type ProtectedVolumeReplicationGroupListStatus struct {
```

Comment:
After mulling over this problem of creating an API for information from an operator (for a long time), I think the following is a more suitable option, as there is no desired/actual state in the API, it is more a `get` routine.

We should publish the information in a `ConfigMap` that is operator generated and can be used by the admin to know which applications are protected and required data regarding the same. This would be a CM that is used as a read only data regarding the known VRGs a dr-cluster knows from its peer.

Most/all of the reconcile remains the same, but the output is a CM instead of a status in an API.

NOTE: When we pivot to a CM if accepted as a better alternative would be a different discussion.

Reply:
The Kube API conventions mention to a "call-and-response" pattern in the [Spec and Status](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status) section:
> Some objects which are not persisted in the system - such as SubjectAccessReview and other webhook style calls - may choose to add spec and status to encapsulate a "call and response" pattern. The spec is the request (often a request for information) and the status is the response. For these RPC like objects the only operation may be POST, but having a consistent schema between submission and response reduces the complexity of these clients.

It seems to me this protected VRGs API fits this pattern, which uses spec and status for call and response instead of the typical desired and actual states, but it seems valid nonetheless.  Thoughts?

Reply:
I think we are stretching the example to provide a service/app specific API. Put another way we are leveraging existing k8s API server to extend it for a non-k8s resource/API case, that I believe goes against the pattern of use.

The ConfigMap (or some form of data) that the service makes accessible, looks to leverage an existing pattern from k8s instead, and hence seems more appropriate.

Reply:
> The one case where this scheme may work for VRG, is if it returns VRGs in a namesapce (still looking at the code). I.e
> 
>     * User creates this resource in an application namespace
> 
>     * Ramen reconciles to provide VRGs in that namespace as the status
> 
> 
> I think that is the use case here, and not to provide all VRGs across namespaces, in which case this may be a valid option.

If the above is the case, the request should not need the S3Profile, it should list all VRGs from any S3Profile that ramen is configured with. As S3Profile is not something easily available to the user.

Reply:
Issue #544 tracks changing this from a Kube API to a struct stored in a configMap